dandi/girder.py

```diff
@@ -845,3 +844,99 @@ def __exit__(self, exc_type, exc_value, tb):
+
+      - If the default backend *was* the ``EncryptedKeyring`` backend, error.
+
+      - If the ``EncryptedKeyring`` backend is not in the list of available
+        backends (likely because its dependencies are not installed, though
+        that shouldn't happen if dandi was installed properly), error.
+
+      - If ``EncryptedKeyring``'s data file already exists, use it as the
+        backend.
+
+      - If ``EncryptedKeyring``'s data file does not already exist, ask the
+        user whether they want to start using ``EncryptedKeyring``.  If yes,
+        then set ``keyringrc.cfg`` (if it does not already exist) to specify it
+        as the default backend, and return the backend.  If no, error.
+
+    Returns a keyring backend and the password it holds (if any) for the given
+    service and username.
+    """
+
+    from keyring.core import get_keyring, load_config, load_env
+    from keyring.backend import get_all_keyring
+    from keyring.errors import InitError
+    from keyring.util.platform_ import config_root
+    from keyrings.alt.file import EncryptedKeyring
+
+    kb = load_env() or load_config()
+    if kb:
+        return (kb, kb.get_password(service_name, username))
+    kb = get_keyring()
+    try:
+        password = kb.get_password(service_name, username)
+    except InitError as e:
+        lgr.info("Default keyring errors on query: %s", e)
+        if isinstance(kb, EncryptedKeyring):
+            lgr.info(
+                "Default keyring is EncryptedKeyring; abandoning keyring procedure"
+            )
+            raise
+        try:
+            [kb] = [k for k in get_all_keyring() if isinstance(k, EncryptedKeyring)]
+        except ValueError:
+            lgr.info("EncryptedKeyring not available; abandoning keyring procedure")
+            raise e
+        else:
+            if op.exists(kb.file_path):
+                lgr.info("EncryptedKeyring file exists; using as keyring backend")
+                return (kb, kb.get_password(service_name, username))
+            lgr.info("EncryptedKeyring file does not exist")
+            if askyesno("Would you like to establish an encrypted keyring?"):
+                keyring_cfg = Path(config_root()) / "keyringrc.cfg"
+                if keyring_cfg.exists():
+                    lgr.info("%s exists; refusing to overwrite", keyring_cfg)
+                else:
+                    lgr.info(
+                        "Configuring %s to use EncryptedKeyring as default backend",
+                        keyring_cfg,
+                    )
+                    keyring_cfg.parent.mkdir(parents=True, exist_ok=True)
+                    keyring_cfg.write_text(
+                        "[backend]\n"
+                        "default-keyring = keyrings.alt.file.EncryptedKeyring\n"
+                    )
+                return (kb, None)
+            raise
+    else:
+        return (kb, password)
+
+
+def askyesno(question):
```

Comment:
please initiate some helper module, e.g. `dandi/support/ui.py` or alike with this function so it could be used elsewhere when/if needed

Reply:
add support for default?  IMHO in this case `yes` is a reasonable default.  Should be visualized in the prompt as well

Reply:
Done.